wallet/src/components/IbcTransfer.tsx

```diff
@@ -0,0 +1,497 @@
+      disabled={direction === IbcDirection.Deposit}
+      error={!!isRemoteChainAddressInvalid}
+      helperText={
+        isRemoteChainAddressInvalid ? (
+          'Invalid Address'
+        ) : remoteChainBalance !== null ? (
+          <>
+            Balance Available:{' '}
+            {purse?.currentAmount &&
+              purse?.displayInfo &&
+              stringifyPurseValue({
+                value: remoteChainBalance,
+                displayInfo: purse?.displayInfo,
+              })}
+          </>
+        ) : (
+          'Fetching balance...'
+        )
+      }
+      InputProps={{
+        type: 'text',
+        autoComplete: 'off',
+        endAdornment: isRemoteChainAddressInvalid && (
+          <InputAdornment position="end">
+            <Button size="small" onClick={() => connectWithKeplr()}>
+              Use Keplr
+            </Button>
+          </InputAdornment>
+        ),
+      }}
+    />
+  );
+
+  const recipient = (() => {
+    switch (direction) {
+      case IbcDirection.Deposit:
+        return agoricAddressInfo;
+      case IbcDirection.Withdrawal:
+        return remoteChainAddressInput;
+    }
+  })();
+
+  const isAmountInputDisabled =
+    direction === IbcDirection.Deposit && remoteChainBalance === null;
+
+  return (
+    <>
+      <Dialog open={!!purse} onClose={close}>
+        <DialogTitle>
+          IBC Transfer {titlePreposition[direction]}{' '}
+          {ibcAsset?.chainInfo.chainName}
+        </DialogTitle>
+        <DialogContent>
+          {sender}
+          <div style={{ height: '68px' }}>
+            <TextField
+              error={!isAmountInputDisabled && isAmountInvalid}
+              margin="dense"
+              autoFocus={direction === IbcDirection.Withdrawal}
+              label="Send amount"
+              fullWidth
+              variant="standard"
+              value={amount}
+              helperText={
+                !isAmountInputDisabled && isAmountInvalid
+                  ? 'Invalid amount'
+                  : ''
+              }
+              onChange={handleAmountChange}
+              disabled={isAmountInputDisabled}
+              InputProps={{
+                type: 'text',
+                autoComplete: 'off',
+                endAdornment: (
+                  <InputAdornment position="end">
+                    <Petname name={purse?.brandPetname} />
+                  </InputAdornment>
+                ),
+              }}
+            />
+          </div>
+          <Box sx={{ marginY: 1, display: 'flex', justifyContent: 'center' }}>
+            <ArrowDownward />
+          </Box>
+          {recipient}
+          <Typography color="primary">{error}</Typography>
+        </DialogContent>
+        <DialogActions>
+          {inProgress ? (
+            <CircularProgress size={36} sx={{ p: 1, mx: 1 }} />
+          ) : (
+            <>
+              {/* @ts-expect-error 'cancel' is part of our theme */}
```

Comment:
time to put it into the types?

Reply:
I'm not sure, we'd have to wrap the component and override the MUI types somewhere down the inheritance tree. It's a big pain that MUI doesn't allow custom color names.